cmd/cli/cli.go

```diff
@@ -202,25 +202,40 @@ func sourceWriteFunc(t targetiface.Target, ft failureiface.Failure, tr transform
 		transformed := tr(messages)
 		// no error as errors should be returned in the failures array of TransformationResult
 
-		// Ack filtered messages with no further action
+		// Send message buffer
+		messagesToSend := transformed.Result
 		messagesToFilter := transformed.Filtered
-		for _, msg := range messagesToFilter {
-			if msg.AckFunc != nil {
-				msg.AckFunc()
+
+		if cfg.Data.DebugMode {
+			// Make 'Data' field nil for filtered data (we still have original data in a separate field)
+			// This can be used to differentiate data at the target level.
+      // Very ugly. Maybe another boolean field in message to mark it as filtered...?
```

Comment:
Agree that it's ugly. 

What I was thinking of was something like you suggest here - but as I think it through, that way also ends up being ugly.

This outline implementation is good food for thought on how to design that though